app/src/main/java/org/openobservatory/ooniprobe/activity/MainActivity.java

```diff
@@ -272,18 +273,24 @@ public void update(Observable observable, Object data) {
     }
 
     public void checkResources() {
-        SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(this);
-        if (!preferences.getBoolean("resources_copied", false)) {
-            copyResources(R.raw.hosts, "hosts.txt");
-            copyResources(R.raw.geoipasnum, "GeoIPASNum.dat");
-            copyResources(R.raw.geoip, "GeoIP.dat");
-            copyResources(R.raw.global, "global.txt");
-            PreferenceManager.getDefaultSharedPreferences(this).edit().putBoolean("resources_copied", true).apply();
-        }
+        copyResources(R.raw.hosts, "hosts.txt");
+        copyResources(R.raw.geoipasnum, "GeoIPASNum.dat");
+        copyResources(R.raw.geoip, "GeoIP.dat");
+        copyResources(R.raw.global, "global.txt");
     }
 
     private void copyResources(int id, String filename) {
-        Log.v(TAG, "copyResources...");
+        boolean exists = false;
+        try {
+            openFileInput(filename);
+            exists = true;
+        } catch (FileNotFoundException exc) {
+            /* FALLTHROUGH */
+        }
+        if (exists) {
+            return;
+        }
```

Comment:
Yep, initially it was written as you say, then I erred on the side of readability!